packages/@react-spectrum/color/test/HexColorField.test.js

```diff
@@ -0,0 +1,223 @@
+
+    hexColorField.focus();
+    fireEvent.change(hexColorField, {target: {value: 'xyz'}});
+    expect(hexColorField.value).toBe('xyz');
+    expect(onChangeSpy).not.toHaveBeenCalled();
+
+    hexColorField.blur();
+    expect(hexColorField.value).toBe('#AABBCC');
+  });
+
+  it.each`
+    Name                                | expected      | action
+    ${'increment with arrow up key'}    | ${'#AAAAAE'}  | ${(el) => fireEvent.keyDown(el, {key: 'ArrowUp'})}
+    ${'increment with page up key'}     | ${'#AAAAAE'}  | ${(el) => fireEvent.keyDown(el, {key: 'PageUp'})}
+    ${'increment with mouse wheel'}     | ${'#AAAAAE'}  | ${(el) => fireEvent.wheel(el, {deltaY: -10})}
+    ${'decrement with arrow down key'}  | ${'#AAAAA6'}  | ${(el) => fireEvent.keyDown(el, {key: 'ArrowDown'})}
+    ${'decrement with page down key'}   | ${'#AAAAA6'}  | ${(el) => fireEvent.keyDown(el, {key: 'PageDown'})}
+    ${'decrement with mouse wheel'}     | ${'#AAAAA6'}  | ${(el) => fireEvent.wheel(el, {deltaY: 10})}
+  `('should handle $Name event', function ({expected, action}) {
+    renderComponent({defaultValue: '#aaa', step: 4});
+    const hexColorField = screen.getByLabelText('Primary Color');
+    expect(hexColorField.value).toBe('#AAAAAA');
+
+    action(hexColorField);
+    expect(hexColorField.value).toBe(expected);
+  });
+
+  it.each`
+    Name                                 | props                                                    | initExpected  | action
+    ${'not increment beyond max value'}  | ${{defaultValue: '#bbbbba', maxValue: '#bbb', step: 4}}  | ${'#BBBBBA'}  | ${(el) => fireEvent.keyDown(el, {key: 'ArrowUp'})}
+    ${'not decrement beyond min value'}  | ${{defaultValue: '#bbbbbc', minValue: '#bbb', step: 4}}  | ${'#BBBBBC'}  | ${(el) => fireEvent.keyDown(el, {key: 'ArrowDown'})}
+    ${'increment to max value'}          | ${{defaultValue: '#aaa', maxValue: '#bbb'}}              | ${'#AAAAAA'}  | ${(el) => fireEvent.keyDown(el, {key: 'End'})}
+    ${'decrement to min value'}          | ${{defaultValue: '#ccc', minValue: '#bbb'}}              | ${'#CCCCCC'}  | ${(el) => fireEvent.keyDown(el, {key: 'Home'})}
+  `('should $Name', function ({props, initExpected, action}) {
+    renderComponent(props);
+    const hexColorField = screen.getByLabelText('Primary Color');
+    expect(hexColorField.value).toBe(initExpected);
+
+    action(hexColorField);
+    expect(hexColorField.value).toBe('#BBBBBB');
+  });
+
+  it.each`
+    Name            | props                                        | initExpected  | newValue  | action
+    ${'max value'}  | ${{defaultValue: '#aaa', maxValue: '#bbb'}}  | ${'#AAAAAA'}  | ${'fff'}  | ${(el) => fireEvent.change(el, {target: {value: 'fff'}})}
+    ${'min value'}  | ${{defaultValue: '#ccc', minValue: '#bbb'}}  | ${'#CCCCCC'}  | ${'000'}  | ${(el) => fireEvent.change(el, {target: {value: '000'}})}
+  `('should clamp value to $Name on change', function ({props, initExpected, newValue, action}) {
```

Comment:
currently once the color is typed and before the blur happens, onChange is fired with the clamped value, we should verify that onChange call as well

Reply:
Resolved. Added an onChange check in this test